tmt/steps/provision/testcloud.py

```diff
@@ -509,6 +511,10 @@ def start(self):
             self.image_url = self._guess_image_url(self.image_url)
             self.debug(f"Guessed image url: '{self.image_url}'", level=3)
 
+        # Make sure image_url is specified
+        if self.image_url is None:
+            raise ProvisionError("image_url is not found")
```

Comment:
* `image_url` is initialized with `image`, `self.image_url = self.image`, and `self.image` is not expected to be `None`
* if `self.image_url` is not URL-like`, `self.image_url = self._guess_image_url(...)` which raises `ProvisionError` if URL is `None`-ish.

So the question here is, how can `image_url` be `None`?

Reply:
Well, we should not check `image_url` in `GuestTestcloud.start()`.

Reply:
> So the question here is, how can image_url be None?

Yes, good question. I believe the issue has been fixed by using the `SerializableContainer` which correctly saves all keys into `guests.yaml` so that all of them are correctly set during `wake()` up. Thus, closing. Feel free to reopen if you still see the problem.